### PR TITLE
Feature/ch9151/label size editor choice measurement

### PR DESCRIPTION
**Pull request by Godmartinz**
# Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context, providing screenshots where practical. List any dependencies that are required for this change.

Fixes # (issue)

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Test A
- [ ] Test B

**Test Configuration**:
* PHP version:
* MySQL version
* Webserver version
* OS version

# Checklist:

- [ ] I have read the Contributing documentation available here: https://snipe-it.readme.io/docs/contributing-overview
- [ ] I have formatted this PR according to the project guidelines: https://snipe-it.readme.io/docs/contributing-overview#pull-request-guidelines
- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Comment by Godmartinz** [comment]
adds the options of centimeters, millimeters for measurements sizes to labels.
image
image
image

**Comment by Godmartinz** [comment]
Just realized..this is not completed. I have not accounted for measurements in the less file.

**Comment by Godmartinz** [comment]
Do we still want to add this @uberbrady @snipe ?

File: database/migrations/2021_07_27_191709_adds_label_measurement_type.php [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+    /**
+     * Run the migrations.
+     *
+     * @return void
+     */
+    public function up()
+    {
+        Schema::table('settings', function (Blueprint $table) {
+            $table->string('labels_measurement_type')->after('labels_per_page')->default('in');
+            $table->string('old_labels_measurement_type')->after('labels_measurement_type');
```
Comment (uberbrady) [inline database/migrations/2021_07_27_191709_adds_label_measurement_type.php]: I think you're trying to re-add the rest of the settings table here.

**Review by uberbrady — CHANGES_REQUESTED** [review]
One thing I don't quite get yet is the `old_labels_measurement_type` - is that for when people switch from one measurement type to another? I figure you might be able to perform the conversion without it, though, using some way of referring to the 'old' measurement type, but not necessarily storing it in the database.

Barring that, I think in general I'm good with this, but I think there's some repetition that you can clear out - 

One thing that definitely sticks out is that we have some 'magic numbers' interspersed here throughout - probably better to throw those into a variable, right? Then just use that variable.

Another thing is that we see a lot of:

```
$setting->labels_blah = ($setting->label_blah) * $something;
```

And then that repeats over and over again - do you think there might be a way to squeeze out some of that repetition too?

Let me know when you can give some of those a stab - you also mention some Less files in the comments, so if you can address that - either stating that you don't really need them, or that you do, then including those changes - that'd be helpful too. Thanks!

File: database/migrations/2021_07_27_191709_adds_label_measurement_type.php (hunk 1 above)
Comment (Godmartinz) [reply database/migrations/2021_07_27_191709_adds_label_measurement_type.php]: These adjustments are actually necessary changes to the precision of the decimal being stored. the values blow up  if I dont adjust the precision here.

File: app/Http/Controllers/SettingsController.php [hunk 2]
```diff
@@ -856,8 +856,12 @@ public function postLabels(Request $request)
         $setting->labels_pageheight = $request->input('labels_pageheight');
         $setting->labels_display_company_name = $request->input('labels_display_company_name', '0');
         $setting->labels_display_company_name = $request->input('labels_display_company_name', '0');
+        $setting->labels_measurement_type     = $request->input('labels_measurement_type', 'in');
```
Comment (uberbrady) [inline app/Http/Controllers/SettingsController.php:860]: I feel like you can start with a line *above* here saying something like:

```php
$old_measurement_type = $settings->labels_measurement_type;
```

File: app/Http/Controllers/SettingsController.php [hunk 3]
```diff
@@ -856,8 +856,12 @@ public function postLabels(Request $request)
...
 
+        if($setting->labels_measurement_type != $request->old('labels_measurement_type')) {
```
Comment (uberbrady) [inline app/Http/Controllers/SettingsController.php]: And then *here* you can do - 
```php
if($old_measurement_type != $request->labels_measreument_type) {
```
And work your magic in there?

File: app/Http/Controllers/SettingsController.php [hunk 4]
```diff
@@ -896,6 +900,113 @@ public function postLabels(Request $request)
...
+        public function measurementConverter($setting, Request $request)
+        {
+            \Log::info('This is the old setting:'.$request->old('labels_measurement_type'));
+            switch ($setting->labels_measurement_type) {
+
+                case "in":
+//                      converts centimeters to inches
+                    if ($request->old($setting->labels_measurement_type) == 'cm') {
+
+                        $setting->labels_width = ($setting->labels_width) / 2.54;
```
Comment (uberbrady) [inline app/Http/Controllers/SettingsController.php]: And definitely you should really fix this "magic constant" problem - generally we don't like using "magic numbers" - you should use a variable instead. `$cm_per_inch` = 2.54, and then do your division by `$cm_per_inch`.

Once you make that change I think you're going to see another optimization we can do...

**Review by uberbrady — CHANGES_REQUESTED** [review]
See if you can try some of those cleanups I'm suggesting?

**Comment by Godmartinz** [comment]
actually this is not ready to go.

**Comment by chksome** [comment]
Snipe requested that I look at your work on labels to make sure I wasn't duplicating effort/creating conflicts with the pull request that I closed in #10973. I'm assuming this PR is what they were referencing. If so, would you like me to open a PR against your fork/branch, open a new PR, or would you just like to add it? My PR was messy and my change hard to find. In case you'd like to incorporate, I've attached a patch. No attribution required.
[remove_unnecessary_label_css.txt](https://github.com/snipe/snipe-it/files/8586510/remove_unnecessary_label_css.txt)

**Pull request closed without merging**